tests/cpp11_condition_variable/Makefile

```diff
@@ -13,6 +13,12 @@ BOARD_INSUFFICIENT_MEMORY := i-nucleo-lrwan1 nucleo-f334r8 spark-core \
 # to CXXEXFLAGS variable
 CXXEXFLAGS += -std=c++11
 
+# nucleo-f303k8 doesn't have enough RAM to run the test so we reduce the stack
```

Comment:
Can we just add the board to `BOARD_INSUFFICIENT_MEMORY` ?
Or maybe other boards in this variable could benefit from reducing the stacksize (same comment for other tests).

Reply:
Ok, I'll go with the second option, but I won't have access to some of the hardware for a couple of days.

Reply:
Still doesn't work for `i-nucleo-lrwan1` and `stm32l0538-disco`

```
2019-09-20 12:48:18,205 - INFO # main(): This is RIOT! (Version: 2019.10-devel-822-g64bd4-pr_nucleo_f303k8_tests)
2019-09-20 12:48:18,206 - INFO # 
2019-09-20 12:48:18,210 - INFO # ************ C++ condition_variable test ***********
2019-09-20 12:48:18,213 - INFO # Wait with predicate and notify one ... 
2019-09-20 12:48:18,214 - INFO # Done
2019-09-20 12:48:18,214 - INFO # 
2019-09-20 12:48:18,216 - INFO # Wait and notify all ...
```

Reply:
I have plated around with this a bit and it seems the values need to be tweaked quite a lot for it to work. I'm thinking of going back to your first suggestion and keeping the `thread_cooperation` change. what do you think?

Reply:
> what do you think?

Let's keep the PR as it is now.